packages/twenty-server/src/modules/connected-account/auto-companies-and-contacts-creation/services/create-company-and-contact.service.ts

```diff
@@ -89,9 +86,7 @@ export class CreateCompanyAndContactService {
     const filteredContactsToCreate = uniqueContacts.filter(
       (participant) =>
         !alreadyCreatedContactEmails.includes(participant.handle) &&
-        participant.handle.includes('@') &&
-        (isContactAutoCreationForNonWorkEmailsEnabled ||
-          isWorkEmail(participant.handle)),
+        participant.handle.includes('@'),
```

Comment:
Shouldn't we also check if excludeNonProfessionalEmails is true ? (see comment below)